src/UrlGenerator.php

```diff
@@ -136,10 +136,16 @@ public function generateFromCurrent(array $replacedArguments, string $fallbackRo
             throw new RuntimeException('Current route is not detected.');
         }
 
+        $queryParameters = [];
+        if ($this->currentRoute->getUri() !== null) {
+            parse_str($this->currentRoute->getUri()->getQuery(), $queryParameters);
+        }
+
         /** @psalm-suppress PossiblyNullArgument Checked route name on null above. */
         return $this->generate(
             $this->currentRoute->getName(),
-            array_merge($this->currentRoute->getArguments(), $replacedArguments)
+            array_merge($this->currentRoute->getArguments(), $replacedArguments),
```

Comment:
Maybe add a test for `array_merge()` right away to prevent appearance of a new escaped mutant (see warning below)?